weave-js/src/common/components/WandbLoader.tsx

```diff
@@ -50,26 +54,35 @@ export const TrackedWandbLoader = ({
   profilingCb,
   samplingRate = 0.1,
   track,
+  onComplete,
+  onStart,
   ...props
 }: TrackedWandbLoaderProps) => {
-  useLifecycleProfiling(name, (data: ProfileData) => {
-    try {
-      // log the lifecycle for each loader to segment
-      const additionalData = profilingCb ? profilingCb() : {};
-      const trackedData = {
-        componentId: data.id,
-        duration: data.duration,
-        ...additionalData,
-      };
-      const randomNum = Number(Math.random().toString().slice(-2)); // take the last two digits off a random number
-      if (randomNum <= samplingRate * 100) {
-        track('wandb-loader-onscreen', trackedData);
+  useLifecycleProfiling(
+    name,
+    (data: ProfileData) => {
+      try {
+        // log the lifecycle for each loader to segment
+        const additionalData = profilingCb ? profilingCb() : {};
+        const trackedData = {
+          componentId: data.id,
+          duration: data.duration,
+          ...additionalData,
+        };
+        if (onComplete) {
+          onComplete(name, trackedData);
+        }
+        const randomNum = Number(Math.random().toString().slice(-2)); // take the last two digits off a random number
+        if (randomNum <= samplingRate * 100) {
+          track('wandb-loader-onscreen', trackedData);
+        }
+      } catch (e) {
+        // Tracking should be able to fail gracefully without breaking the app
+        captureException?.(e);
       }
-    } catch (e) {
-      // Tracking should be able to fail gracefully without breaking the app
-      captureException?.(e);
-    }
-  });
+    },
+    onStart
```

Comment:
i found this confusing, having a huge function as another function's argument, and then another random one line argument tacked on below it.

i think we should a) useCallback for the second argument, and b) change useLifecycleProfiling to use named arguments instead of positional arguments. i think named arguments are better in general, and should be required if a function has 3 or more args. (don't forget to memo the arg object if necessary though.)